tests/nd_item/nd_item_async_work_group_copy.cpp

```diff
@@ -33,111 +23,65 @@ class TEST_NAME : public util::test_base {
-                                             stride);
-              });
-        });
-
-        queue.submit([&](cl::sycl::handler &cgh) {
-          auto accGlobal =
-              buf.get_access<cl::sycl::access::mode::read_write>(cgh);
-
-          auto accLocal =
-              cl::sycl::accessor<size_t, 1, cl::sycl::access::mode::read_write,
-                                 cl::sycl::access::target::local>(
-                  cl::sycl::range<1>(BUFFER_SIZE), cgh);
-
-          // Test 3D
-          cgh.parallel_for<class nd_item_async_work_group_copy_3d>(
-              cl::sycl::nd_range<3>(
-                  cl::sycl::range<3>(RANGE_SIZE_1D, RANGE_SIZE_2D,
-                                     RANGE_SIZE_3D),
-                  cl::sycl::range<3>(1, 1, 1)),
-              [=](cl::sycl::nd_item<3> ndItem) {
-                auto ptrGlobal = accGlobal.get_pointer();
-                auto ptrLocal = accLocal.get_pointer();
-
-                ndItem.async_work_group_copy(ptrLocal, ptrGlobal, BUFFER_SIZE);
-                ndItem.async_work_group_copy(ptrGlobal, ptrLocal, BUFFER_SIZE);
-
-                constexpr size_t stride = 2;
-                constexpr size_t numElements = BUFFER_SIZE / stride;
-                ndItem.async_work_group_copy(ptrLocal, ptrGlobal, numElements,
-                                             stride);
-                ndItem.async_work_group_copy(ptrGlobal, ptrLocal, numElements,
-                                             stride);
-              });
-        });
-      }
-
-      queue.wait_and_throw();
-
-    } catch (const cl::sycl::exception &e) {
-      log_exception(log, e);
-      cl::sycl::string_class errorMsg =
-          "a SYCL exception was caught: " + cl::sycl::string_class(e.what());
-      FAIL(log, errorMsg.c_str());
-    }
+    check_type<size_t>(log);
+    check_type_and_vec<bool>(log);
+    check_type_and_vec<char>(log);
+    check_type_and_vec<signed char>(log);
+    check_type_and_vec<unsigned char>(log);
+    check_type_and_vec<short int>(log);
+    check_type_and_vec<unsigned short int>(log);
+    check_type_and_vec<int>(log);
+    check_type_and_vec<unsigned int>(log);
+    check_type_and_vec<long int>(log);
+    check_type_and_vec<unsigned long int>(log);
+    check_type_and_vec<long long int>(log);
+    check_type_and_vec<unsigned long long int>(log);
+    check_type_and_vec<float>(log);
+
+    check_type_and_vec<cl::sycl::byte>(log);
+    check_type_and_vec<cl::sycl::cl_bool>(log);
+    check_type_and_vec<cl::sycl::cl_char>(log);
+    check_type_and_vec<cl::sycl::cl_uchar>(log);
+    check_type_and_vec<cl::sycl::cl_short>(log);
+    check_type_and_vec<cl::sycl::cl_ushort>(log);
+    check_type_and_vec<cl::sycl::cl_int>(log);
+    check_type_and_vec<cl::sycl::cl_uint>(log);
+    check_type_and_vec<cl::sycl::cl_long>(log);
+    check_type_and_vec<cl::sycl::cl_ulong>(log);
+    check_type_and_vec<cl::sycl::cl_float>(log);
+
+#ifdef INT8_MAX
```

Comment:
I appreciate the pedantism and the micro-optimization in this area.
I guess it might be difficult to test the tests if we do not have 9-bit host computer, for example... :-)
But is it possible to have `std::is_same<cl::sycl::cl_char, std::int8_t>::value` false according to the spec?
In that case we should not try to run a test but just `static_assert` at least say it is inconsistent.

Reply:
Thank you for comments and sorry for the delay.
According to the spec cl_char is "Alias to a signed 8-bit integer, as defined by the C++11 ISO Standard" and from my understanding it doesn't guaranteed that it's the same type as std::int8_t, even if it's likely, that's why I think we can't skip tests for intN_t types without these checks. And since they are optional in C++11 Standard it's not guaranteed that they are defined, that's why #ifdef INT8_MAX is used - to make sure int8_t is defined. 

 Do you think it would be better to just run tests for intN_t types if they are defined?